beep/protocol_tools/maccor_to_arbin.py

```diff
@@ -0,0 +1,407 @@
+# Copyright 2019 Toyota Research Institute. All rights reserved.
+""" Parsing and conversion of maccor procedure files to arbin schedule files"""
+
+import os
+import re
+from datetime import datetime
+from beep import PROTOCOL_SCHEMA_DIR
+from beep.protocol_tools.arbin_schedule_file import ScheduleFile
+from collections import OrderedDict
+from monty.serialization import loadfn
+
+TEST_DIR = os.path.dirname(__file__)
+TEST_FILE_DIR = os.path.join(TEST_DIR, "test_files")
+
+
+class ProcedureToSchedule:
+    def __init__(self, procedure_dict_steps,
+                 version=None
+                 ):
+        self.service = version
+        self.procedure_dict_steps = procedure_dict_steps
+
+    def create_sdu(self, sdu_input_name, sdu_output_name):
+        """
+        Highest level function in the class. Takes a schedule file and replaces
+        all of the steps with steps from the procedure file. Then writes the
+        resulting schedule file to the output file.
+
+        Args:
+            sdu_input_name (str): the full path of the schedule file to use as a
+                shell for the steps
+            sdu_output_name (str): the full path of the schedule file to output
+
+        """
+        schedule = ScheduleFile()
+        sdu_dict = schedule.to_dict(sdu_input_name)
+
+        keys = list(sdu_dict['[Schedule]'].keys())
+        for key in keys:
+            if '[Schedule' in key:
+                del sdu_dict['[Schedule]'][key]
+
+        step_name_list, step_flow_ctrl = self.create_metadata()
+        for step_index, step in enumerate(self.procedure_dict_steps):
+            step_arbin = self.compile_to_arbin(self.procedure_dict_steps[step_index],
+                                               step_index, step_name_list, step_flow_ctrl)
+            key = '[Schedule_Step{}]'.format(step_index)
+            sdu_dict['[Schedule]'].update({key: step_arbin})
+        schedule.dict_to_file(sdu_dict, sdu_output_name)
+
+    def create_metadata(self):
+        """
+        Creates the necessary information so that flow control for the steps can be
+        set up correctly.
+
+        Returns:
+            list: unique strings consisting of the step number and note
+            dict: key value pairs for the step that
```

Comment:
Doctring for the return dict needs completion.

Reply:
Yup, must have forgotten to finish that thought...